[import-w3c-tests] Fix and Improve the examples for the --help command

karlcow opened the pull request:
#### 8303df68dcd382c4fbff2adc2939d79b9669de7e
<pre>
[import-w3c-tests] Fix and Improve the examples for the --help command
<a href="https://bugs.webkit.org/show_bug.cgi?id=274819">https://bugs.webkit.org/show_bug.cgi?id=274819</a>
<a href="https://rdar.apple.com/128925292">rdar://128925292</a>

Reviewed by Anne van Kesteren.

The current help menu is a bit confusing, this is an attempt at making
clearer for everyone who is using a local source folder for the tests.
With reviews and suggestions from Marcos Caceres and Anne van Kesteren

* Tools/Scripts/webkitpy/w3c/test_importer.py:

Canonical link: <a href="https://commits.webkit.org/279499@main">https://commits.webkit.org/279499@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e767f751905935cd197fa20acdc7379dad7c0a99

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53649 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33010 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6164 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56930 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4375 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55952 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/40495 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/4220 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/43465 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2857 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55746 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/31226 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46384 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24599 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/53489 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/28058 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3701 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/2530 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/49787 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3887 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/58524 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28811 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/3933 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/50873 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/30011 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46551 "") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/50215 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11691 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30943 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29788 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

karlcow commented on Tools/Scripts/webkitpy/w3c/test_importer.py:
```diff
@@ -121,8 +121,13 @@ def format(self, record):
+    import-w3c-tests web-platform-tests/xyz
+
+To import a web-platform-tests suite from a specific local folder:
+   1. Your folder containing WPT needs to be called "web-platform-tests/"
```
self fix: 
```suggestion
   1. Your folder containing the Web platform tests needs to be called "web-platform-tests/"
```

annevk replied:
Perhaps:

> Your checkout of web-platform-tests/wpt needs to have the folder name web-platform-tests/

karlcow replied:
Done.

marcoscaceres commented on Tools/Scripts/webkitpy/w3c/test_importer.py:
```diff
@@ -121,8 +121,13 @@ def format(self, record):
-To import a web-platform-tests test suite named xyz, use: 'import-w3c-tests web-platform-tests/xyz'.
-To import a web-platform-tests suite from a specific folder, use 'import-w3c-tests xyz -s my-folder-containing-web-platform-tests-folder'"""
+To import a web-platform-tests test suite named xyz, use:
+    import-w3c-tests web-platform-tests/xyz
```
You also want tip of tree, right? I think there's already multiple bugs filed that it should default to tip-of-tree

```suggestion
To import a web-platform-tests test suite named xyz, use:
    import-w3c-tests --tip-of-tree web-platform-tests/xyz
```

karlcow replied:
done. Thanks.

marcoscaceres commented on Tools/Scripts/webkitpy/w3c/test_importer.py:
```diff
@@ -121,8 +121,13 @@ def format(self, record):
+To import a web-platform-tests suite from a specific local folder:
+   1. Your checkout of "web-platform-tests/wpt" needs to have the folder name "web-platform-tests/"
+   2. If the path is "~/dev/web-platform-tests/", use:
+      import-w3c-tests web-platform-tests/xyz -s ~/dev/  --clean-dest-dir"""
```
Let's use the " --src-dir" form, so it's more clear: 

```suggestion
To import a web-platform-tests suite from a specific local checkout out of web platform tests:
   1. Your checkout must be a folder called "web-platform-tests".
   2. If the path is "~/dev/web-platform-tests/", use:
      import-w3c-tests web-platform-tests/xyz --src-dir ~/dev/  --clean-dest-dir"""
```

karlcow replied:
done. Thanks

annevk reviewed (approved):
cc @youennf for awareness

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer.py:
```diff
@@ -121,8 +121,13 @@ def format(self, record):
+    import-w3c-tests --tip-of-tree web-platform-tests/xyz
+
+To import a web-platform-tests suite from a specific local checkout out of web platform tests:
+   1. Your checkout must be a folder called "web-platform-tests".
```
```suggestion
   1. Your working tree must be in a directory called "web-platform-tests".
```

(per https://support.apple.com/en-gb/guide/applestyleguide/apsg7af4f5d0/web#apd11f2025f4c0b4, "In developer materials, use directory"; git's docs mostly use "working tree" and not checkout)

marcoscaceres replied:
"working tree" is confusing IMO. That's not colloquial terminology. I think "checkout" is more appropriate because one does "git **checkout** git:some-repo"

karlcow replied:
yes I have difficulties to understand the new sentence proposed here. 

What about:

1. Your local WPT copy must be in a directory called  "web-platform-tests".

karlcow replied:
To make it clearer

```suggestion
   1. Your local WPT copy must be in a directory called "web-platform-tests".
```

karlcow replied:
To the point that @marcoscaceres is making, the change is to try to make the help easy to understand for anyone. And indeed colloquial is then probably more important in this case.

marcoscaceres replied:
I like @karlcow's wording.

karlcow replied:
done.

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer.py:
```diff
@@ -121,8 +121,13 @@ def format(self, record):
+To import a web-platform-tests test suite named xyz, use:
+    import-w3c-tests --tip-of-tree web-platform-tests/xyz
+
+To import a web-platform-tests suite from a specific local checkout out of web platform tests:
```
```suggestion
To import a web-platform-tests suite from an existing working tree of web-platform-tests:
```

marcoscaceres replied:
as above... at least, I've personally never heard anyone call a checkout a "working tree".

karlcow replied:
```suggestion
To import a web-platform-tests suite from a local copy of web platform tests:
```

karlcow replied:
done.

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer.py:
```diff
@@ -121,8 +121,13 @@ def format(self, record):
+
+To import a web-platform-tests suite from a specific local checkout out of web platform tests:
+   1. Your checkout must be a folder called "web-platform-tests".
+   2. If the path is "~/dev/web-platform-tests/", use:
```
```suggestion
   2. If the working tree is at "~/dev/web-platform-tests", use:
```

karlcow replied:
```suggestion
   2. If the local copy is at "~/dev/web-platform-tests/", use:
```

marcoscaceres replied:
I would say "If the local copy is at, for example, ...".

But note the even the "~" is a bit of Unix thing... it's not a Windows thing as far as I know.

karlcow replied:
done.

marcoscaceres reviewed (approved):
Looks great @karlcow! thank you so much!!! 🥰

karlcow committed:
[import-w3c-tests] Fix and Improve the examples for the --help command
https://bugs.webkit.org/show_bug.cgi?id=274819
rdar://128925292

Reviewed by Anne van Kesteren.

The current help menu is a bit confusing, this is an attempt at making
clearer for everyone who is using a local source folder for the tests.
With reviews and suggestions from Marcos Caceres and Anne van Kesteren

* Tools/Scripts/webkitpy/w3c/test_importer.py:

Canonical link: https://commits.webkit.org/279499@main
